src/material-experimental/mdc-slider/slider-thumb-decorator.ts

```diff
@@ -0,0 +1,74 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  ElementRef,
+  EventEmitter,
+  Input,
+  Output,
+  ViewChild,
+  ViewEncapsulation,
+} from '@angular/core';
+import {Thumb} from '@material/slider';
+
+/**
+ * Handles displaying the slider knobs and their value indicators.
+ */
+@Component({
+  selector: 'mat-slider-start-thumb-decorator, mat-slider-end-thumb-decorator',
```

Comment:
At least to me, "decorator" implies that it can be mixed together with other decorators, but since this is an element tag, it can't really be mixed. Is there a reason why we don't just call it `mat-slider-start-thumb, mat-slider-end-thumb`?

Reply:
The `MatSliderThumb` Directive (which wraps the sliders' hidden `input` element(s)) already uses the selector `mat-slider-thumb`, which might make it a bit confusing.

We could rename `MatSliderThumb` to `MatSliderInput` and use the selector `mat-slider-input` instead. I believe this was the initial plan, but @jelbourn or @mmalerba wanted it named `mat-slider-thumb` to be more intuitive for developers. That is also when it was decided that the visual slider thumb class would be named `MatSliderThumbDecorator` and have the selector `mat-slider-thumb-decorator`.

Reply:
I didn't so much agree to naming it "decorator" as agreed with the general approach and the naming of the components which constitute the user facing API. I do agree that this name is kind of confusing because it sounds like it might have something to do with the [decorator pattern](https://en.wikipedia.org/wiki/Decorator_pattern).

My understanding is that this is a purely internal component, so my first question is, does this component even need to exist at all or can we just roll it into the mat-slider class? If we do think it deserves its own component maybe something like `mat-slider-thumb-ornament`?

Reply:
The main reason I moved it into a separate component was to dedup the html for the start and end thumbs. I can combine it with `MatSlider` by doing something like this instead:
```html
<ng-container *ngIf="isRange">
  <ng-container *ngIf="isDiscrete">
    <div class="mdc-slider__value-indicator-container">
      <div class="mdc-slider__value-indicator">
        <span class="mdc-slider__value-indicator-text">
          {{ startValueIndicatorText }}
        </span>
      </div>
    </div>
  </ng-container>
  <div class="mdc-slider__thumb-knob" #startKnob></div>
</ng-container>

<ng-container *ngIf="isDiscrete">
  <div class="mdc-slider__value-indicator-container">
    <div class="mdc-slider__value-indicator">
      <span class="mdc-slider__value-indicator-text">
        {{ endValueIndicatorText }}
      </span>
    </div>
  </div>
</ng-container>
<div class="mdc-slider__thumb-knob" #endKnob></div>
```

Otherwise, I'm good with renaming it to `MatSliderThumbOrnament` with the selectors `mat-slider-start-thumb-ornament` and `mat-slider-end-thumb-ornament`. Let me know your thoughts

Reply:
Ah ok, you could probably eliminate the need to copy paste with a `*ngFor="let thumb of thumbs"`. Though I'd also be fine having it extracted into a separate component. I'll leave it up to you, whatever you feel makes it easier.

Reply:
I would recommend not introducing a separate component since that incurs extra mental overhead vs. just using a for loop